TxnLab/use-wallet
# feat: Adding new mnemonic wallet provider for e2e testing

### Proposed changes:

- New provider called mnemonic, similar to kmd it prompts user for 25 word passphrase that upon entering is stored as an account record with addr and pk - passphrase itself isn't persisted.
- Adding prettierrc file to make sure folks contributing and using prettier won't have conflicts with indent style used in codebase

### How to test?

Refer to the forked main branch - execute `yarn install` and `yarn dev` to launch the playbook. Make sure to explicitly add it to playbook because similarly to kmd its excluded from default list of providers. 


### Who may benefit from this provider?

Any folks who has integration with use-wallet and would like to have a simple way to perform e2e test, with mnemonic provider like this user can automate pasting the phrase to the input prompt in most automation libraries out there. In contrast e2e test with real walletconnect/myalgowallet or etc gets tricker as the amount of complexity to automate it is far greater.

### Further notes

There were no unit tests setup in the repo so instead i tested it locally so far - will use this forked version in my swapper.algoworld.io project for the time being to have a more specific justification on why this can be helpful. Feel free to leave any comments/suggestions - would love to address those
